### DISCUSSION TITLE
Perform actions by client app-id/title with `niri msg`

### DISCUSSION THREAD

**Discussion by e-tho** [YaLTeR/niri#507] [Ideas]
This would allow me to focus my browser after running a search in my web menu script using fuzzel.

**Comment by YaLTeR** [comment]
This is something that will eventually happen one way or another, but will need more thought and then development

**Comment by sommerper** [comment]
This would also help me immensely because I'm used to focus my most used apps with direct shortcuts.

**Comment by e-tho** [comment]
@YaLTeR,
Btw, my use case seems to be related to the lack of support for the xdg-activation protocol rather than this specific feature. I noticed that while it has been implemented, some windows don't grab focus as they should.

**Reply by YaLTeR** [reply]
Yeah, the "one app activates web browser" kind of case works if the client uses xdg-activation (which is implemented in niri).
